src/platform/plugins/shared/discover/public/application/main/state_management/redux/actions/save_discover_session.ts

```diff
@@ -66,6 +66,7 @@ export const saveDiscoverSession = createInternalStateAsyncThunk(
     >();
 
     let nextSelectedTabId = state.tabs.unsafeCurrentId;
+    const selectedTab = currentTabs.find((tab) => tab.id === state.tabs.unsafeCurrentId);
```

Comment:
```suggestion
    const selectedTab = selectTab(state, state.tabs.unsafeCurrentId)
```
Nit: we could use `selectTab` for this.

Reply:
Yeah, but using `currentTabs` directly seemed more straightforward and without the extra logic of `selectTab` implementation.

Reply:
I guess I'm just partial to using the selectors when possible, but logically it's the same so all good either way.